### ISSUE TITLE
Should esoteric cop errors be fixed? 

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12783]
I've been doing some "fuzzing" on the RuboCop repo the last few days. My intention with this to catch errors that are likely to be causes while typing code with an LSP enabled since I've encountered a few of those myself while using `ruby-lsp` in VSCode.

The ""fuzzing"" involves running RuboCop on each permutation of the test cases that are available. But I'm running out of things that I would actually qualify as an issue.

`Lint/EmptyLinesAroundAttributeAccessor` for example raises on this code:
```rb
attr_reader :foo
# Some comment
```

I don't think anyone would write that realistically since it must appear at the top-level. I would produce a PR for the fix myself but I'm wondering if things like this would just be a waste of time for maintainers.

**Comment by bbatsov** [comment]
I think we probably shouldn't go overboard with this. If something is unlikely to happen in practice we can ignore it. 

> The ""fuzzing"" involves running RuboCop on each permutation of the test cases that are available.

I was curious how you discovered so many weird issues! Now I know. :-) Thanks for working on this!

**Comment by Earlopain** [comment]
> I think we probably shouldn't go overboard with this. If something is unlikely to happen in practice we can ignore it.

I suspected something like that, thanks. I'll just let these things be, they can be fixed if someone else stumbles over it naturally.

Perhaps I'll make a few more PRs here in a similar vein but the majority of realistic issues is already done.
